src/main/java/com/ita/if103java/ims/service/impl/WarehouseServiceImpl.java

```diff
@@ -66,20 +94,39 @@ public WarehouseDto findWarehouseById(Long id) {
     }
 
     @Override
-    public List<WarehouseDto> findWarehousesByTopLevelId(Long topLevelId) {
+    public List<WarehouseDto> findWarehousesByTopLevelId(Long topLevelId, UserDetailsImpl user) {
         return warehouseDtoMapper.toDtoList(warehouseDao.findChildrenByTopWarehouseID(topLevelId));
     }
 
     @Override
-    public WarehouseDto update(WarehouseDto warehouseDto) {
+    public WarehouseDto update(WarehouseDto warehouseDto, UserDetailsImpl user) {
         Warehouse updatedWarehouse = warehouseDtoMapper.toEntity(warehouseDto);
         Warehouse dBWarehouse = warehouseDao.findById(updatedWarehouse.getId());
         updatedWarehouse.setActive(dBWarehouse.isActive());
+        EventName eventName = EventName.WAREHOUSE_EDITED;
+        createEvent(user, updatedWarehouse, eventName);
```

Comment:
Can be simplified
createEvent(user, warehouse, EventName.WAREHOUSE_EDITED);

Please add event message "Updated Warehouse '{name}' with level '{level}'"
For example:
Updated Warehouse 'Storage' with level '0'